.github/workflows/build.yml

```diff
@@ -7,8 +7,32 @@ on:
       - litmus-v2
 
 jobs:
+  changes:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      frontend: ${{ steps.filter.outputs.frontend }}
+    steps:
+    # For pull requests it's not necessary to checkout the code
+    - uses: dorny/paths-filter@v2
+      id: filter
+      with:
+        filters: |
+          frontend:
+            - 'litmus-portal/frontend/**'
+          graphql-server:
+            - 'litmus-portal/graphql-server/**'
+          authentication:
+            - 'litmus-portal/authentication/**'
+          event-tracker:
+            - 'litmus-portal/cluster-agents/event-tracker/**'
+          subscriber:
+            - 'litmus-portal/cluster-agents/subscriber/**'
+
   backend-checks:
     runs-on: ubuntu-latest
+    needs: changes
+    if: needs.changes.outputs.graphql-server == 'true' || needs.changes.outputs.authentication == 'true' || needs.changes.outputs.authentication == 'true' || needs.changes.outputs.event-tracker == 'true' || needs.changes.outputs.subscriber == 'true'
```

Comment:
You can access the outputs of a step using id of that step with this format - `steps.<id>.outputs.<variable>`
e.g. 
`steps.filter.outputs.graphql-server` 
cc: @uditgaurav PTAL if it is good or should be changed.